execute/exectypes/outcome.go

```diff
@@ -89,51 +89,31 @@ func (o Outcome) ToLogFormat() Outcome {
 	for i, report := range o.CommitReports {
 		commitReports[i] = report.CopyNoMsgData()
 	}
-	reports := make([]cciptypes.ExecutePluginReportSingleChain, len(o.Report.ChainReports))
-	for i, report := range o.Report.ChainReports {
-		reports[i] = report.CopyNoMsgData()
+	truncatedReports := make([]cciptypes.ExecutePluginReport, len(o.Reports))
+	for i, execReport := range o.Reports {
+		truncatedReports[i].ChainReports = make([]cciptypes.ExecutePluginReportSingleChain, len(execReport.ChainReports))
+		for j, report := range execReport.ChainReports {
+			truncatedReports[i].ChainReports[j] = report.CopyNoMsgData()
+		}
 	}
 	cleanedOutcome := Outcome{
 		State:         o.State,
 		CommitReports: commitReports,
-		Report: cciptypes.ExecutePluginReport{
-			ChainReports: reports,
-		},
+		Reports:       truncatedReports,
 	}
 	return cleanedOutcome
 }
 
 // NewOutcome creates a new Outcome with the pending commit reports and the chain reports sorted.
+// No sorting is needed, they are already in a canonical representation from the builder.
 func NewOutcome(
 	state PluginState,
 	selectedCommits []CommitData,
-	report cciptypes.ExecutePluginReport,
-) Outcome {
-	return NewSortedOutcome(state, selectedCommits, report)
-}
-
-// NewSortedOutcome ensures canonical ordering of the outcome.
-// TODO: handle canonicalization in the encoder.
-func NewSortedOutcome(
-	state PluginState,
-	pendingCommits []CommitData,
-	report cciptypes.ExecutePluginReport,
+	reports []cciptypes.ExecutePluginReport,
 ) Outcome {
-	pendingCommitsCP := append([]CommitData{}, pendingCommits...)
-	reportCP := append([]cciptypes.ExecutePluginReportSingleChain{}, report.ChainReports...)
-	sort.Slice(
-		pendingCommitsCP,
-		func(i, j int) bool {
-			return LessThan(pendingCommitsCP[i], pendingCommitsCP[j])
-		})
-	sort.Slice(
-		reportCP,
-		func(i, j int) bool {
-			return reportCP[i].SourceChainSelector < reportCP[j].SourceChainSelector
-		})
 	return Outcome{
 		State:         state,
-		CommitReports: pendingCommitsCP,
-		Report:        cciptypes.ExecutePluginReport{ChainReports: reportCP},
```

Comment:
Are we sure about removing these sorts? IIRC they were there for a reason

Reply:
Moderately sure. At least in the report builder (for the filter phase), it should return data in a canonical format. Maybe the getMessages / getCommitReports functions could still use this, but I think business logic around the ordering belongs in those functions.